src/input/fpcap.c

```diff
@@ -91,6 +92,16 @@ static int _open(input_fpcap_t* self)
 {
     mlassert_self();
 
+#if _POSIX_C_SOURCE >= 200112L || defined(__FreeBSD__)
+    if (self->use_fadvise) {
+        int err = posix_fadvise(fileno((FILE*)self->file), 0, 0, POSIX_FADV_SEQUENTIAL);
+        if (err) {
+            lcritical("posix_fadvise() failed: %s", core_log_errstr(err));
```

Comment:
Unfortunately treating this as a hard error is problematic. It can fail for any number of reasons which are not "critical", e.g. on named pipes (`mkfifo`). At the same time it's hard for the Lua caller to know if a file is fifo.

I suggest to treat this as truly "advisory", log failure and continue on. It would be much easier to use.

Reply:
critical is not a hard error.

And it's up to the user to use fadvise or not, if you use it on the wrong stuff then it's the users fault.

Reply:
Right, I should have commented on the next line. `lcritical` is not problem by itself, but the `ret -2` is causing breakage.

As for "user's fault", I can see your point, but still think this API is not user friendly.

`fpcap` module already requires two branches to open file, and with this file it would require third:

```
if stdin ... use openfp
else ... enable fadvise ... open ... if it fails try open again without fadvise
```

If you want to have a strict mode, can you please add a relaxed mode which it's not a hard error? E.g. as `Fpcap:fadvise_sequential()` parameter or so?

Reply:
Sorry but no, I would expect an API to return an error on an error, not lie about it.

If you don't care then:
```
err = input:open()
if err == -2 then
  input.obj.use_fadvise = 0
  err = input:open()
end
```